meson.build

```diff
@@ -151,6 +157,29 @@ else
   ucx_dep = dependency('ucx', modules: ['ucx::ucs', 'ucx::ucp', 'ucx::uct'])
 endif
 
+# UCX GPU device API detection
+ucx_gpu_device_api_available = false
+if ucx_dep.found() and cuda_dep.found()
```

Comment:
why need to check cuda_dep.found()? 
the presence of ucp/api/device/ucp_device_impl.h header already means UCX has device API